api-2.2/src/main/java/org/openmrs/module/coreapps/htmlformentry/EncounterDiagnosesElement.java

```diff
@@ -378,10 +378,16 @@ public void applyAction(FormEntrySession formEntrySession) {
     public Set<org.openmrs.Diagnosis> getExistingDiagnoses(FormEntryContext context) {
         FormEntryContext.Mode mode = context.getMode();
         if (mode == FormEntryContext.Mode.EDIT || mode == FormEntryContext.Mode.VIEW) {
-        	Encounter existingEncounter = context.getExistingEncounter();
-        	if (existingEncounter != null) {
-        		return existingEncounter.getDiagnoses();
-        	}
+            Encounter existingEncounter = context.getExistingEncounter();
+            if (existingEncounter != null) {
+                Set<org.openmrs.Diagnosis> nonVoidedDiagnoses = new HashSet<org.openmrs.Diagnosis>();
+                for (org.openmrs.Diagnosis diagnosis : existingEncounter.getDiagnoses()) {
+                    if (!diagnosis.getVoided()) {
+                        nonVoidedDiagnoses.add(diagnosis);
+                    }
+                }
+                return nonVoidedDiagnoses;
```

Comment:
I was hoping we could do this:
```java
return existingEncounter.getDiagnoses().stream().filter(d -> !d.getVoided()).collect(Collectors.toSet());
```
But for this to work we would need to put the JDK compliance to 1.8 (_just_ for the api-2.2 submodule). Do you think you could try that out?

Reply:
I tried just to specify JDK 1.8 in the api-2.2 module but, all existing tests relying Spring ApplicationContext began to fail. It seems to be using Spring 3 thus failing to load Java 8 classes.

```
java.lang.IllegalStateException: Failed to load ApplicationContext
(...)
Caused by: org.springframework.core.NestedIOException: ASM ClassReader failed to parse class file - probably due to a new Java class file version that isn't supported yet
```

Reply:
@mks-d, api-2.2 submodule is now using JDK 8 and I've reviewed the code to use streams.